lib/ide/ast/impl/src/lib.rs

```diff
@@ -775,42 +791,42 @@ Layer<T> for WithID<S> {
     }
 }
 
-impl<T> HasSpan for WithID<T>
-where T: HasSpan {
-    fn span(&self) -> usize {
-        self.deref().span()
+impl<T> HasSize for WithID<T>
+where T: HasSize {
+    fn size(&self) -> usize {
+        self.deref().size()
     }
 }
 
 
-// === WithSpan ===
+// === WithSize ===
 
 /// Stores a value of type `T` and information about its span.
 ///
 /// Even if `T` is `Spanned`, keeping `span` variable is desired for performance
 /// purposes.
 #[derive(Eq, PartialEq, Debug, Shrinkwrap, Serialize, Deserialize)]
 #[shrinkwrap(mutable)]
-pub struct WithSpan<T> {
+pub struct WithSize<T> {
     #[shrinkwrap(main_field)]
     #[serde(flatten)]
     pub wrapped: T,
-    pub span: usize
+    pub size: usize
```

Comment:
👎

Reply:
Especially in Rust, where `Sized` trait means that the memory size is known. In Rust size is bound to memory

Reply:
Then my suggestion would be `len()` or `length()`.
Definitely not span, it means `from<->to` where both bear value, like https://doc.rust-lang.org/proc_macro/struct.Span.html

Reply:
I don't think "size should be bound to memory" is a general rule of naming things.
But I think `length` is a better name.
And I would add, that `WithSpan` is silly in this context, because with span of what? Span definition is 
> the full extent of something from end to end; the amount of space that something covers.
What covers something what implements "WithSpan" trait?

Reply:
I like `Length` and `HasLength` . Lets do it